acceptance/repository-cloudant/docker.setup.js

```diff
@@ -3,314 +3,81 @@
-    const port = _.get(c, [
-      'NetworkSettings',
-      'Ports',
-      '5984/tcp',
-      '0',
-      'HostPort',
-    ]);
-    process.env.CLOUDANT_PORT = port;
-    process.env.CLOUDANT_HOST = host;
-    const usr = process.env.CLOUDANT_USERNAME;
-    const pass = process.env.CLOUDANT_PASSWORD;
-    process.env.CLOUDANT_URL =
-      'http://' + usr + ':' + pass + '@' + host + ':' + port;
-    console.log(
-      'env:',
-      _.pick(process.env, [
-        'CLOUDANT_URL',
-        'CLOUDANT_HOST',
-        'CLOUDANT_PORT',
-        'CLOUDANT_USERNAME',
-        'CLOUDANT_PASSWORD',
-        'CLOUDANT_DATABASE',
-      ]),
-    );
-    next(null, container);
-  });
+async function db() {
+  await createDB();
+  await logStatus();
 }
 
-function waitFor(path) {
-  return function waitForPath(container, next) {
-    const opts = {
-      host: process.env.CLOUDANT_HOST,
-      port: process.env.CLOUDANT_PORT,
-      path: path,
-    };
-
-    console.log(`waiting for instance to respond: ${opts}`);
-    return ping(null, CONNECT_RETRIES);
-
-    function ping(err, tries) {
-      console.log('ping (%d/%d)', CONNECT_RETRIES - tries, CONNECT_RETRIES);
-      if (tries < 1) {
-        next(err || new Error('failed to contact Cloudant'));
-      }
-      http
-        .get(opts, function(res) {
-          res.pipe(devNull());
-          res.on('error', tryAgain);
-          res.on('end', function() {
-            if (res.statusCode === 200) {
-              setImmediate(next, null, container);
-            } else {
-              tryAgain();
-            }
-          });
-        })
-        .on('error', tryAgain);
-      // eslint-disable-next-line no-shadow
-      function tryAgain(err) {
-        setTimeout(ping, CONNECT_DELAY, err, tries - 1);
-      }
-    }
+async function createDB() {
+  const usr = process.env.CLOUDANT_USERNAME;
+  const pass = process.env.CLOUDANT_PASSWORD;
+  const host = process.env.CLOUDANT_HOST;
+  const port = process.env.CLOUDANT_PORT;
+  process.env.CLOUDANT_URL =
+    'http://' + usr + ':' + pass + '@' + host + ':' + port;
+
+  const opts = {
+    method: 'PUT',
+    path: '/' + process.env.CLOUDANT_DATABASE,
+    host: process.env.CLOUDANT_HOST,
+    port: process.env.CLOUDANT_PORT,
+    auth: process.env.CLOUDANT_USERNAME + ':' + process.env.CLOUDANT_PASSWORD,
   };
-}
-
-function createAdmin() {
-  return function createAdminUser(container, next) {
-    const data = `"${process.env.CLOUDANT_PASSWORD}"`;
-    const uri =
-      '/_node/couchdb@127.0.0.1/_config/admins/' +
-      process.env.CLOUDANT_USERNAME;
-    const opts = {
-      method: 'PUT',
-      path: uri,
-      header: {
-        'Content-Type': 'application/x-www-form-urlencoded',
-      },
-      host: process.env.CLOUDANT_HOST,
-      port: process.env.CLOUDANT_PORT,
-      body: data,
-    };
 
-    const req = http.request(opts, function(res) {
+  http
+    .request(opts, function(res) {
```

Comment:
Would it be simpler to use an module such as `axios`?

Reply:
I tried it a bit but it's not that easy to convert to `axios` :see_no_evil:. I left a comment to indicate that it can be improved by `axios`.